docs/overview.rst

```diff
@@ -1,18 +1,106 @@
+
+.. code:: python
+
+   from ethpm import Package
+   from eth_utils import to_canonical_address
+
+   @pytest.fixture
+   def owned_deployer(solc_deployer):
+       owned_manifest_path = Path('to/owned/manifest.json')
+       owned_deployer = solc_deployer(path=owned_manifest_path)
+       return owned_deployer.deploy("Owned")
+
+   def test_owned_contract(owned_deployer)
+       owned_package, owned_address = owned_deployer
+       assert isinstance(owned_package, Package) 
+       owned_contract_instance = owned_deployer.deployments.get_deployment_instance("Owned")
+       assert to_canonical_address(owned_contract_instance.address) == owned_address
+
+
+.. py:method:: Deployer.deploy(contract_type)
+
+   Deploys an instance of given `contract_type` if sufficient data is present in the manifest.
+
+.. py:method:: Deployer.register_strategy(contract_type, strategy)
+
+   If a `contract_type` requires linking, then you *must* register a valid strategy constructed with the ``Linker`` before you can deploy an instance of the `contract_type`.
+
+
+Linker
+------
+
+If a contract factory requires linking, you must register a "strategy" for a particular contract factory with the deployer. It is up to you to design an appropriate strategy for a contract factory. 
+
+Two ``linker`` functions are made available:
+
+.. py:method:: deploy(contract_name, *args=None)
+
+   To deploy an instance of `contract_name`. If the contract constructor requires arguments, they must also be passed in.
+
+.. py:method:: link(contract_name, linked_type)
+
+   Links a `contract_name` to a `linked_type`. The `linked_type` must have already been deployed.
 
-To deploy any of these contracts onto the default ``w3`` instance, simply call ``deploy`` on the deployer and a ``w3.contract.Contract`` instance will be returned representing the ``Package`` instance the contract type belongs to along with the address of the newly deployed contract type.
+For example, the `Escrow` contract factory requires linking to an instance of the `SafeSendLib` before an `Escrow` contract instance can be deployed. This is how you would set up a strategy for `Escrow`
 
 .. code:: python
+  
+   @pytest.fixture
+   def escrow_deployer(solc_deployer, w3, manifest_dir):
+       escrow_manifest_path = manifest_dir / "escrow_manifest.json"
+       return solc_deployer(escrow_manifest_path), w3
 
-    @pytest.fixture
-    def contract_name(deployer):
-        return deployer.deploy("ContractName")
 
+   @pytest.fixture
+   def escrow_contract_instance(escrow_deployer):
+       deployer, w3 = escrow_deployer
+       escrow_strategy = linker(
+           deploy("SafeSendLib"),
+           link("Escrow", "SafeSendLib"),
+           deploy("Escrow", w3.eth.accounts[0]),
+       )
+       deployer.register_strategy("Escrow", escrow_strategy)
+       escrow_deployer = deployer.deploy("Escrow")
```

Comment:
name nitpick, `escrow_deployment` or maybe `escrow_deployment_data`.  I was thoroughly confused until I read the next line of code.

Maybe just update to be something like the following since you don't actually make use of the address.

```python
linked_escrow_package, _ = deployer.deploy("Escrow")
```